go.mod

```diff
@@ -1,34 +1,31 @@
 module github.com/fluxcd/go-git-providers
 
-go 1.20
+go 1.21
 
 require (
-	code.gitea.io/sdk/gitea v0.16.0
+	code.gitea.io/sdk/gitea v0.17.0
```

Comment:
While we're at it we should probably upgrade the gitea version used for the e2e tests.